interpreter/nodev8/v8.go

```diff
@@ -1877,6 +1935,26 @@ func (d *v8Data) readIntrospectionData(ef *pfelf.File, syms libpf.SymbolFinder)
 	}
 
 	// Add some defaults when needed
+	if d.version >= v8Ver(11, 9, 0) {
+		// the class hierarchy changed: HeapObject no longer
+		// derives from Object. This confuses gen-postmortem-metadata.py
+		// and it no longer emits class hierarchy information correctly.
+		// But ScopeInfo indeed still derives from HeapObject, so just set
+		// that manually here.
+		vms.ScopeInfo.HeapObject = true
+	}
+	if d.version >= v8Ver(12, 5, 0) && !vms.DeoptimizationData.FixedArray &&
+		!vms.DeoptimizationData.TrustedFixedArray {
+		vms.DeoptimizationData.ProtectedFixedArray = true
+	} else if d.version >= v8Ver(12, 3, 0) && !vms.DeoptimizationData.FixedArray {
+		// DeoptimizationData changed base type to TrustedFixedArray, which doesn't have metadata.
+		vms.DeoptimizationData.TrustedFixedArray = true
+	}
+	if d.version >= v8Ver(12, 4, 0) && !vms.SourcePositionTable.ByteArray {
+		// SourcePositionTable changed base type to TrustedByteArray, which doesn't have metadata.
+		vms.SourcePositionTable.TrustedByteArray = true
+	}
```

Comment:
I have generally tried really hard to not put any version comparisons in filling the metadata. Instead try to figure out a means to detect the from other fields if its possible. I assume this is not easily possible as its needed to disambiguate the base type here.

Reply:
Right, it wasn't possible here unfortunately.